src/Analysis/Ast/Impl/Values/Scope.cs

```diff
@@ -42,7 +42,7 @@ internal class Scope : IScope {
         public IScope OuterScope { get; }
         public bool VisibleToChildren { get; }
 
-        public IReadOnlyList<IScope> Children => _childScopes ?? Array.Empty<IScope>() as IReadOnlyList<IScope>;
+        public IReadOnlyList<IScope> Children => _childScopes?.ToArray() ?? Array.Empty<IScope>();
```

Comment:
`_childScopes` looks like a good candidate for `ImmutableArray`, cause the child scope is never removed. And it can be safely returned from the `Children` property as is. But it's up to you.